### DISCUSSION TITLE
Allow previous notes on the timeline to be seen in the gameplay field in the osu!catch editor

### DISCUSSION THREAD

**Discussion by Wrongwayscout** [ppy/osu#25170] [Ideas]
I'm no expert catch mapper, but I've tried to use the new catch editor a few times and it has always stuck out to me that you are unable to see notes prior to your timeline cursor's position in the gameplay field. They disappear as soon as the notes reach the catcher.
I find this to be quite limiting, as it encourages you to have your timeline cursor earlier than where you're placing notes, so that you can have a complete picture of the patterns you are creating.
I think it would be beneficial for catch to do something similar to the stable mania editor, and have previous notes stay visible even after they pass the judgement line/catcher.

Current catch editor:
![Screenshot 2023-10-19 012545](https://github.com/ppy/osu/assets/84840821/15ec68fa-4b10-4dd5-96b5-b060c45a1c16)

In stable mania it looks like this (green line is judgement line)
![Screenshot 2023-10-19 012121 2](https://github.com/ppy/osu/assets/84840821/f08da336-9743-4be2-a0ba-bc54fb726aca)

Here's what it would look like in the catch editor (height exaggerated):
![Screenshot 2023-10-19 012545 2](https://github.com/ppy/osu/assets/84840821/255bace1-ce86-48cd-bfa9-7e10bf6d1057)

This somewhat reflects how things are in the stable catch/standard editor which does have notes stick around after their timeline position has passed.

**Comment by peppy** [comment]
Sounds valid.
